### ISSUE TITLE
Parse failure on Java file

### ISSUE THREAD

**Issue by stil4m** [thorbenvh8/chevrotain-java#23]
I've encountered a Java parse error while using `prettier-java` on a private project.

The small scale example looks like this:

```java
public class MyClass {

    public void foo() {
        String.toString('');
    }

    public void bar() {
        String.toString(String.toString(''));
    }
}
```

The error received in the command line is:

```
[error] Test.java: Error: Sad sad panda, parsing errors detected in line: 8, column: 44!
[error] Expecting --> '}' <-- but found --> ')' <--
[error]     at Error (native)
[error]     at Object.parse (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier-plugin-java/node_modules/chevrotain-java/src/index.js:20:11)
[error]     at Object.parse (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier-plugin-java/src/parser.js:6:25)
[error]     at Object.parse$2 [as parse] (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:7002:19)
[error]     at coreFormat (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:10198:23)
[error]     at format (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:10369:16)
[error]     at formatWithCursor (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:10381:12)
[error]     at Object.formatWithCursor (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:31224:15)
[error]     at format$1 (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33227:21)
[error]     at /Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33413:16
[error]     at /Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33359:14
[error]     at Array.forEach (native)
[error]     at eachFilename (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33358:15)
[error]     at Object.formatFiles (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33374:3)
[error]     at Object.run (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33949:14)
[error]     at Object.<anonymous> (/Users/me/.nvm/versions/node/v6.14.3/lib/node_modules/prettier/bin-prettier.js:33964:5)
```

Versions:

* prettier: `1.13.7`
* prettier-java: `0.0.10`

I can create a regression test, but I'm not sure where to start in the repository. If you can give me some pointers, I maybe can create a PR.

**Comment by bd82** [comment]
There is additional context information on the error objects.
I suggest you breakpoint on the throw and inspect the error object's context property.
It has the following API: https://sap.github.io/chevrotain/documentation/3_7_0/interfaces/irecognizercontext.html

That will give you a hint where in the parser execution the error occurred so you can place break points **directly** in the parser.

In addition because you have the position of the error you can try to minimize the input
to the most minimal input that reproduces the bug.
That will also give you a strong hint regarding where in the parser the error occurred.

**Comment by stil4m** [comment]
The input is already as minimal as possible. Removing one of the methods of method invocations resolves the error.

I'll look into the context property of the error.

**Comment by stil4m** [comment]
So I've looked into the issue more.

The example is actually fixed in the `0.0.19` version of chevrotain-java. The problem exists in the installation of `prettier-plugin-java` on my machine. I'll create an issue in that repository.
